src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/MacroProcessor.cs

```diff
@@ -0,0 +1,44 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Core.Contracts;
+using Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Abstractions;
+using Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Macros.Config;
+
+namespace Microsoft.TemplateEngine.Orchestrator.RunnableProjects
+{
+    internal static class MacroProcessor
+    {
+        // Warning: if there are unknown macro "types", they are quietly ignored here.
+        // This applies to both the regular and deferred macros.
+        internal static void ProcessMacros(
+            IEngineEnvironmentSettings environmentSettings,
+            GlobalRunConfig runConfig,
+            IVariableCollection variables)
+        {
+            foreach (BaseMacroConfig config in runConfig.ComputedMacros)
+            {
+                config.Evaluate(environmentSettings, variables);
```

Comment:
do we want to try-catch all errors and continue if macro evaluation fails?

Reply:
added better error handling in last commit. 
Unknown macro is a warning for now.